spec/omniauth/realme_spec.rb

```diff
@@ -1,7 +1,51 @@
 # frozen_string_literal: true
 
-RSpec.describe Omniauth::Realme do
+RSpec.describe OmniAuth::Realme do
   it 'has a version number' do
-    expect(Omniauth::Realme::VERSION).not_to be nil
+    expect(OmniAuth::Realme::VERSION).not_to be nil
+  end
+
+  describe '.generate_metadata_xml' do
+    let(:idp_metadata_path) { File.join(__dir__, '../fixtures/realme_mts_idp_metadata.xml') }
+    let(:assertion_consumer_service_url) { 'http://www.example.com/auth/anything' }
+    let(:issuer) { 'Anything' }
+    let(:realme_strategy_options) do
+      p12 = OpenSSL::PKCS12.new(File.read(File.join(__dir__, '../fixtures/mts_saml_sp.p12')), 'password')
+
+      {
+        idp_service_metadata: idp_metadata_path,
+        issuer: issuer,
+        private_key: p12.key.to_s,
+        certificate: p12.certificate.to_s,
+        assertion_consumer_service_url: assertion_consumer_service_url,
+        allowed_clock_drift: 0
+      }
+    end
+
+    let(:expected_xml) do
```

Comment:
This string will become wrong when/if PR #12 lands so I will update this PR after that.

Reply:
Done.